java/arcs/core/host/ExplicitHostRegistry.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.core.host
+
+import arcs.core.sdk.Particle
+import kotlin.reflect.KClass
+
+/**
+ * A HostRegistry that discovers the available [ArcHost]s available on this platform by using
+ * explicitly registered [ArcHost]s and [Particle]s invoked by [HostRegistry.registerHost] and
+ * [ExplicitHostRegistry.registerParticles].
+ */
+class ExplicitHostRegistry() : AnnotationBasedHostRegistry() {
+
+    companion object {
+        val instance: ExplicitHostRegistry by lazy(
+            LazyThreadSafetyMode.PUBLICATION
+        ) { ExplicitHostRegistry() }
+    }
+
+    /**
+     * Explicitly register all particles used.
+     */
+    fun registerParticles(allParticles: List<KClass<out Particle>>) {
+        hosts.map { host ->
+            registerParticles(findParticlesForHost(allParticles, host), host)
+        }
+    }
+}
```

Comment:
Rather than rolling your own Singleton, consider using `object` to let Kotlin do it for you.

This would then become just:

```kotlin
object ExplicitHostRegistry : AnnotationBasedHostRegistry() {
    fun registerParticles(allParticles: List<KClass<out Particle>>) {
        // ...
    }
}
```

This way when you use the registry, you won't need to use `.instance` everywhere.